packages/flutter/lib/src/gestures/drag.dart

```diff
@@ -54,31 +54,50 @@ typedef void GesturePanEndCallback(Velocity velocity);
 /// [GesturePanDownCallback] did not complete.
 typedef void GesturePanCancelCallback();
 
-typedef void _GesturePolymorphicUpdateCallback<T>(T delta);
+/// Signature for when a pointer that is in contact with the screen and moving
+/// has moved again. For one-dimensional drags (e.g., horizontal or vertical),
+/// T is `double`, as in [GestureDragUpdateCallback]. For two-dimensional drags
+/// (e.g., pans), T is `Offset`, as in GesturePanUpdateCallback.
+typedef void GesturePolymorphicUpdateCallback<T>(T delta);
 
 bool _isFlingGesture(Velocity velocity) {
   assert(velocity != null);
   final double speedSquared = velocity.pixelsPerSecond.distanceSquared;
   return speedSquared > kMinFlingVelocity * kMinFlingVelocity;
 }
 
-abstract class _DragGestureRecognizer<T extends dynamic> extends OneSequenceGestureRecognizer {
+/// Recognizes movement.
+///
+/// In contrast to [MultiDragGestureRecognizer], [DragGestureRecognizer]
+/// recognizes a single gesture sequence for all the pointers it watches, which
+/// means that there the recognizer has started at most one drag sequence at any
```

Comment:
Yeah, I went through a bunch of drafts for that text and wasn't happy with the result.  I'm trying to say that you'll never see `start`, `start`, `end`, `end` because the first drag sequence hasn't ended by the time the second one wants to start.  Instead, they'll get merged together, which means you'll see the first start and the last end.  It's confusing to explain.  :(